### PR TITLE
VFR_HUD: add attitude for user display

### PR DESCRIPTION
**Pull request by jlecoeur**
( Requires MAVLink PR https://github.com/mavlink/mavlink/pull/1247 )

This PR handles the extension of VFR_HUD with attitude fields meant for user display.

See MAVLink PR for details.

fixes https://github.com/PX4/Firmware/issues/5291 https://github.com/PX4/Firmware/issues/10405

### CODE REVIEW

**Comment by dagar** [comment]
Seems fine to me as long as no one is opposed to the mavlink change.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -1429,6 +1441,29 @@ class MavlinkStreamVFRHUD : public MavlinkStream
...
+				_rotation = get_rot_quaternion(ROTATION_PITCH_90);
+
+			} else {
+				// Normal case
+				_rotation = get_rot_quaternion(ROTATION_NONE);
+			}
+
+			// Attitude
+			const matrix::Quatf att = matrix::Quatf(attitude.q) * _rotation;
+			msg.q1 = att(0);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp]: I suppose this is a question for https://github.com/mavlink/mavlink/pull/1247, but why isn't the quaternion an array?

Comment (jlecoeur) [reply src/modules/mavlink/mavlink_messages.cpp]: I used the same format as in ATTITUDE_QUATERNION.
Both should be array in my opinion, or at least 0 indexed...

File: src/modules/mavlink/mavlink_main.cpp [hunk 2]
```diff
@@ -1635,7 +1635,7 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 		configure_stream_local("SERVO_OUTPUT_RAW_0", 1.0f);
 		configure_stream_local("SYS_STATUS", 1.0f);
 		configure_stream_local("UTM_GLOBAL_POSITION", 0.5f);
-		configure_stream_local("VFR_HUD", 4.0f);
+		configure_stream_local("VFR_HUD", 15.0f);
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_main.cpp]: @dagar remember our conversation? So which one is really used?

Comment (jlecoeur) [reply src/modules/mavlink/mavlink_main.cpp]: In QGC, once ATTITUDE_QUATERNION is received, ATTITUDE is ignored.

Comment (julianoes) [reply src/modules/mavlink/mavlink_main.cpp]: Same in MAVSDK.

**Comment by jlecoeur** [comment]
replaced by https://github.com/PX4/Firmware/pull/13249

**Pull request closed without merging**